extensions/standard-processors/processors/LogAttribute.cpp

```diff
@@ -137,33 +137,32 @@ void LogAttribute::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
       message << "\n" << "Payload:" << "\n";
       ReadCallback callback(flow->getSize());
       session->read(flow, &callback);
-      for (unsigned int i = 0, j = 0; i < callback.read_size_; i++) {
-        message << std::hex << callback.buffer_[i];
-        j++;
-        if (j == 80) {
-          message << '\n';
-          j = 0;
-        }
+
+      auto payload_hex = utils::StringUtils::to_hex(callback.buffer_.data(), callback.buffer_.size());
+      for (size_t i = 0; i < payload_hex.size(); i += 80) {
```

Comment:
Isn't this potentially missing a partial line?

Reply:
Why would it? std::string::substr will take as much as it can, even if the specified length is larger than what remains.
If the string is 40 bytes long, it will run one iteration and create a line of 40 bytes.
If the string is 80 bytes long, it will run one iteration and create a line of 80 bytes.
If the string is 120 bytes long, it will run two iterations, creating a line of 80 bytes, and a line of 40 bytes from offset 80.

Reply:
Sorry I was referring to both interpretations -- but misread the conditional in full view about the max size of the FF and thought it would only log the first MB, but it looks like it only logs flow files < 1 MB  -- which while not clear from the options is at least a sensical line in the sand.